hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/report/OverviewReporter.java

```diff
@@ -0,0 +1,66 @@
+package com.blackducksoftware.integration.hub.detect.workflow.report;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.blackducksoftware.integration.hub.detect.workflow.bomtool.BomToolEvaluation;
+import com.blackducksoftware.integration.hub.detect.workflow.extraction.ReportConstants;
+
+public class OverviewReporter {
+
+    public void writeReport(final ReportWriter writer, final List<BomToolEvaluation> results) {
+        final Map<File, List<BomToolEvaluation>> byDirectory = results.stream()
+                .collect(Collectors.groupingBy(item -> item.getEnvironment().getDirectory()));
+
+        printDirectories(writer, byDirectory);
+
+    }
+
+    private void printDirectories(final ReportWriter writer, final Map<File, List<BomToolEvaluation>> byDirectory) {
+        writer.writeSeperator();
+        for (final File file : byDirectory.keySet()) {
+            final List<BomToolEvaluation> results = byDirectory.get(file);
+
+            final List<BomToolEvaluation> applicable = new ArrayList<>();
+            final List<BomToolEvaluation> extractable = new ArrayList<>();
+            final List<BomToolEvaluation> extractionSuccess = new ArrayList<>();
+            final List<BomToolEvaluation> extractionFailure = new ArrayList<>();
+
+            for (final BomToolEvaluation result : results) {
+                if (result.isApplicable()) {
+                    applicable.add(result);
+                    if (result.isExtractable()) {
+                        extractable.add(result);
+                        if (result.wasExtractionSuccessful()) {
+                            extractionSuccess.add(result);
+                        } else {
+                            extractionFailure.add(result);
+                        }
+                    }
+                }
+            }
```

Comment:
There's a lot of nesting going on here. Would streams be better?

Reply:
Improved.